go/discord-command-bot/main.go

```diff
@@ -0,0 +1,65 @@
+package handler
+
+import (
+	"os"
+
+	"github.com/open-runtimes/types-for-go/v4"
+)
+
+type any = map[string]interface{}
+
+func Main(Context *types.Context) types.ResponseOutput {
+	err := errorIfEnvMissing([]string{
+		"DISCORD_PUBLIC_KEY",
+		"DISCORD_APPLICATION_ID",
+		"DISCORD_TOKEN",
+	})
+	if err != nil {
+		Context.Error(err.Error())
+		return Context.Res.Text("", 500, nil)
+	}
+
+	err = discordVerifyKey(
+		Context.Req.BodyText(),
+		Context.Req.Headers["x-signature-ed25519"],
+		Context.Req.Headers["x-signature-timestamp"],
+		os.Getenv("DISCORD_PUBLIC_KEY"),
+	)
+	if err != nil {
+		Context.Error(err.Error())
+		return Context.Res.Json(any{
+			"error": "Invalid request signature.",
+		}, 401, nil)
+	}
+
+	Context.Log("Valid request")
+
+	discordBody, err := discordParseBody(Context)
+	if err != nil {
+		Context.Error(err.Error())
+		return Context.Res.Json(any{
+			"error": "Invalid body.",
+		}, 400, nil)
+	}
+
+	ApplicationCommandType := 2
+	if discordBody.Type == ApplicationCommandType && discordBody.Data.Name == "hello" {
+		Context.Log("Matched hello command - returning message")
+
+		channelMessageWithSource := 4
+		return Context.Res.Json(
+			any{
```

Comment:
maybe have a struct for this prepared too?

Reply:
Its overkill here I think. Discord response can have MANY different params, and I don't want this to be library that defined it.